src/influxdb_ioxd.rs

```diff
@@ -144,6 +144,14 @@ pub async fn main(config: Config) -> Result<()> {
         server_config
     };
 
+    if config.grpc_bind_address == config.http_bind_address {
+        error!(
+            "grpc and http bind addresses must differ; both are {:?}",
+            config.grpc_bind_address
+        );
```

Comment:
I think you can also do something like this to get structured logging:

```suggestion
        error!(%config.grpc_bind_address, "grpc and http bind addresses must differ");
```

More details can be found in the `tracing` crate docs [here](https://docs.rs/tracing/0.1.26/tracing/#recording-fields).